metrics/metric_observer_test.go

```diff
@@ -8,12 +8,52 @@ import (
 )
 
 func TestPrometheusMetrics(t *testing.T) {
-	problems, err := testutil.GatherAndLint(prometheus.DefaultGatherer)
-	if err != nil {
-		t.Fatal(err)
-	}
+	r := prometheus.NewPedanticRegistry()
+
+	mo := NewMetricObserver(WithRegisterer(r))
+
+	for _, tc := range []struct {
+		name     string
+		updateFn func()
+		expCount int
+	}{
+		{
+			name:     "init",
+			updateFn: func() {},
+			expCount: 1,
+		},
+		{
+			name:     "set routine count to zero",
+			updateFn: func() { mo.RunningRoutineCount(0) },
+			expCount: 1,
+		},
+		{
+			name: "set routine count to 1",
+			updateFn: func() {
+				mo.RunningRoutineCount(1)
```

Comment:
it would be useful to also verify the presence of the `async_routine_manager_routine_instances` metric but I see no way to call `RoutineStarted()` with a mock of `async.AsyncRoutine`.

Reply:
See #10

Reply:
Given https://github.com/openshift-online/async-routine/pull/10 was merged, @simonpasquier do you want to revisit this PR?